app/core/rate_limiting.py

```
@@ -0,0 +1,357 @@
+    async def _get_current_count(self, redis_key: str) -> int:
+        """
+        Retrieve the current request count stored under the given Redis key.
+
+        Returns:
+            int: Current count for the key, or 0 if the key does not exist or an error occurs while reading Redis.
+        """
+        try:
+            redis = db_manager.get_redis()
+            count = await redis.get(redis_key)
+            return int(count) if count else 0
+        except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)